CMakeLists.txt

```diff
@@ -0,0 +1,32 @@
+# CMake build file for Stratum code.
+#
+# Copyright 2022 Intel Corporation
+# SPDX-License-Identifier: Apache 2.0
+
+cmake_minimum_required(VERSION 3.5)
+project(stratum)
+
+set(CMAKE_POSITION_INDEPENDENT_CODE ON)
+
+set(CMAKE_LIBRARY_OUTPUT_DIRECTORY "${CMAKE_INSTALL_PREFIX}/lib")
+set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${CMAKE_INSTALL_PREFIX}/lib")
+
+find_package(absl REQUIRED)
+
+option(BUILD_SHARED_LIBS "Build shared library" OFF)
```

Comment:
This option will allow cmake to build all libraries as static or shared.

Reply:
Hmm. I agree with you in principle. The problem is, I need a little more flexibility to build `infrap4d` because I want to defer responsibility for building and installing the entity that contains Stratum to a higher layer. The most effective way to do this is to build a static library (or libraries) within Stratum and then use the archive to build infrap4d.

Is it essential to your use case that the end product be installed in its final form within Stratum itself, or could you delegate to the upper-layer CMakeLists file that includes Stratum as a subdirectory?

I also have specific requirements with respect to the compiled protobufs. I wish to install libraries and header files that client applications can use to communicate with the Stratum. Have you given much thought to how to use the probobuf libraries in your use case?

Reply:
It is the latter. I would want to package stratum along with the SDE as a separate .so file an application will link to.

My thinking is the current implementation treats Stratum independently. It would also allow it to be integrated as part of a large source tree. Infrap4d for example can actually choose to use the _o objects directly instead of linking libstratum.a if it wants to.

For the protobuf files, I want to generate a _o object file like the existing model and allow them also to be exportable to upper layers.